### ISSUE TITLE
Cannot set properties of undefined (setting 'microsoftTeams')

### ISSUE THREAD

**Issue by qscn** [OfficeDev/microsoft-teams-library-js#1506]
```
fail: Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost[111]
      Unhandled exception in circuit 'WxyS46ZelgIeeKGylLOs9ETBorYsh6NGISsSh_Bzlq0'.
      Microsoft.JSInterop.JSException: Cannot set properties of undefined (setting 'microsoftTeams')
TypeError: Cannot set properties of undefined (setting 'microsoftTeams')
    at https://unpkg.com/@microsoft/teams-js@2.7.0/dist/MicrosoftTeams.min.js:1:221
    at https://unpkg.com/@microsoft/teams-js@2.7.0/dist/MicrosoftTeams.min.js:1:226
         at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
         at SPGConnect.Interop.InteropModuleBase.ImportModuleAsync() in C:\Users\v-shuoqiu\Source\Workspaces\VMS\VMS.NET-branch\SPGConnect\Interop\InteropModuleBase.cs:line 36
         at SPGConnect.Interop.InteropModuleBase.InvokeAsync[T](String functionName, Object[] args) in C:\Users\v-shuoqiu\Source\Workspaces\VMS\VMS.NET-branch\SPGConnect\Interop\InteropModuleBase.cs:line 27
         at SPGConnect.Components.Welcome.OnAfterRenderAsync(Boolean firstRender) in C:\Users\v-shuoqiu\Source\Workspaces\VMS\VMS.NET-branch\SPGConnect\Components\Welcome.razor:line 87
         at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)
```

Hi team,

We got an error without a new deployment. I mean, after the latest deployment, it was working fine at first, but after a few days, we got this error. We haven't made any changes during this time.

Would you please let us know how to solve this error? Thanks in advance.

**Comment by mejlismilanek** [comment]
Hello! Thanks for posting the issue. I'm getting the same error message. I tried reinstalling the SDKs, Visual Studio, and Office, but nothing has helped. It worked fine until today and I didn't change anything either. Regards

**Comment by WayneAASPLLC** [comment]
Hmm, I am also having the same issue.

**Comment by ChetanSharma-msft** [comment]
Hello Everyone,
Hope you are doing well!!

Could you please confirm whether you are facing this issue in Teams JS 2.7.0?
Also, please confirm whether you are using CDN link for Teams JS or imported the Teams JS package?

**Comment by jirka88** [comment]
> I am personally using Microsoft.TeamsFx C#. Both versions 1.1.0 & 1.2.0 do not work and 1.1.0 worked yesterday.

I also use Microsoft.TeamsFx 1.1.0 C# and I have exactly the same problem.

**Comment by mejlismilanek** [comment]
> Hello Everyone, Hope you are doing well!!
> 
> Could you please confirm whether you are facing this issue in Teams JS 2.7.0? Also, please confirm whether you are using CDN link for Teams JS or imported the Teams JS package?

Hello and TY for responding! 

The error message:
_Cannot set properties of undefined (setting 'microsoftTeams')
TypeError: Cannot set properties of undefined (setting 'microsoftTeams')
    at https://unpkg.com/@microsoft/teams-js@2.7.0/dist/MicrosoftTeams.min.js:1:221
    at https://unpkg.com/@microsoft/teams-js@2.7.0/dist/MicrosoftTeams.min.js:1:226_
    
    
I'm using the the Teams FX as well, updating the package didn't help either. It's the JS 2.7.0.

**Comment by ChetanSharma-msft** [comment]
Thanks for sharing the information.

**Comment by ChetanSharma-msft** [comment]
Could you please confirm whether you are facing this issue in Teams Web client or in Teams Desktop client?

**Comment by WayneAASPLLC** [comment]
Due to it being an active dev environment, everything is being ran in Blazor locally, so it uses a website by default.

**Comment by WayneAASPLLC** [comment]
```
    protected override async Task OnAfterRenderAsync(bool firstRender)
    {
        await base.OnAfterRenderAsync(firstRender);

        if (firstRender)
        {
            isInTeams = await MicrosoftTeams.IsInTeams(); //<-----it fails here

            if (isInTeams)
            {
```

**Comment by mejlismilanek** [comment]
> Could you please confirm whether you are facing this issue in Teams Web client or in Teams Desktop client?

I'm getting the error with both the web client in Chrome and the desktop app too. Blazor server Teams tab app run locally while in development.

**Comment by ChetanSharma-msft** [comment]
We will try to repro your scenario via setting below sample and let you know the updates:
https://github.com/OfficeDev/TeamsFx-Samples/blob/dev/graph-toolkit-productivity-hub-aspnetcore/OneProductivityHubAspNetCore/Pages/TabConfig.razor

If you have any other reference sample and steps, please do share.

**Comment by WayneAASPLLC** [comment]
If this is about the tabconfig, i'll give you what I use just in case:
```
@page "/config"
@inject MicrosoftTeams MicrosoftTeams;
@inject NavigationManager NavigationManager;

<div class="config page" id ="tab config">
        <h1>Tab Configuration</h1>
        <p>
            This is where you will add your tab configuration options the user 
            can choose when the tab is added to your team/group chat.
        </p>
</div>

@code {
    private Guid _entityId = Guid.NewGuid();

    protected override async Task OnAfterRenderAsync(bool firstRender)
    {
        if(firstRender)
        {
            var settings = new TeamsInstanceSettings
            {
                //SuggestedDisplayName = "My Tab",
                EntityId = _entityId.ToString(),
                //ContentUrl = $"{NavigationManager.BaseUri}/tab",
                //WebsiteUrl = $"{NavigationManager.BaseUri}/tab"
            };

            await MicrosoftTeams.InitializeAsync();
            await MicrosoftTeams.RegisterOnSaveHandlerAsync(settings);
            //await MicrosoftTeams.InitializeAsync();
        }
    }
}
```

**Comment by noahdarveau-MSFT** [comment]
Hello everyone. We believe this issue is due to a recent webpack change that was included in the recent 2.7.0 release. We are looking to push out a fixed release later today.

**Comment by RTerheggen** [comment]
We are also having same issue

**Comment by noahdarveau-MSFT** [comment]
Hello everyone. I have just released teams-js version 2.7.1 which has removed the webpack change we believe is causing this issue. Please try the latest version and let us know if the issue persists.

**Comment by qscn** [comment]
> Hello everyone. I have just released teams-js version 2.7.1 which has removed the webpack change we believe is causing this issue. Please try the latest version and let us know if the issue persists.

Great! Thanks. I confirm that our app works now.

**Comment by RTerheggen** [comment]
It is working now for us, Thx!

**Comment by RTerheggen** [comment]
> Hello everyone. I have just released teams-js version 2.7.1 which has removed the webpack change we believe is causing this issue. Please try the latest version and let us know if the issue persists.

We are creating a missing critical Teams app, and we have been really scared by this update that made our solution unusable for 2 days.

Questions:
1. Is there a centralized dashboard for monitoring known/current Teams issues / outages? 
2. Are we to expect unannounced changes to Teams infrastructure without any notification
3. Can we know before updates are made? eg patch tuesdays
4. Is there anything we can do defensively to protect us in this situation? Perhaps caching previous versions?

Thanks!

Rob

**Comment by AE-MS** [comment]
> > Hello everyone. I have just released teams-js version 2.7.1 which has removed the webpack change we believe is causing this issue. Please try the latest version and let us know if the issue persists.
> 
> We are creating a missing critical Teams app, and we have been really scared by this update that made our solution unusable for 2 days.
> 
> Questions:
> 
> 1. Is there a centralized dashboard for monitoring known/current Teams issues / outages?
> 2. Are we to expect unannounced changes to Teams infrastructure without any notification
> 3. Can we know before updates are made? eg patch tuesdays
> 4. Is there anything we can do defensively to protect us in this situation? Perhaps caching previous versions?
> 
> Thanks!
> 
> Rob

Thanks for the questions, @RTerheggen. To ensure that the answers are most helpful to you and your situation, can you provide a little information about how your team handles updates and validation of the teams-js library? E.g., how do you consume the library (CDN or npm), how do you look for/get notified of new versions of the library, what is the update timing you use, how do you validate newer versions before updating your solution, etc.

Thank you!
